parser/mpFuncStr.cpp

```diff
@@ -138,6 +138,123 @@ MUP_NAMESPACE_START
     return new FunStrRight(*this);
   }
 
+  //------------------------------------------------------------------------------
+  //
+  // default_value() auxiliary overloading functions
+  //
+  //------------------------------------------------------------------------------
+  int_type default_value(int_type value, int_type standard) {
+    return value == NULL ? standard : value;
+  }
+
+  float_type default_value(float_type value, float_type standard) {
+    return value == NULL ? standard : value;
+  }
+
+  string_type default_value(string_type value, string_type standard) {
+    return value.empty() ? standard : value;
+  }
+
+  // If the value is an integer, it is NULL, therefore return standard
+  string_type default_value(int_type value, string_type standard) {
+    return standard;
+  }
+
+  bool_type default_value(bool_type value, bool_type standard) {
+    return value;
+  }
+
+  // If the value is an integer, it is NULL, therefore return standard
+  bool_type default_value(int_type value, bool_type standard) {
+    return standard;
+  }

```

Comment:
Here @dnovaes 

We don't have `default_value(float_type value, int_type standard)` 👁️ 👁️ 

That's why we were having this problem...

```rb
[10] pry(main)> Parsec::Parsec.eval_equation('default_value(10.0, 0)')
=> 10
​
[11] pry(main)> Parsec::Parsec.eval_equation('default_value(10.5, 0)')
SyntaxError: Can't evaluate function/operator "default_value": Value "10.5" is of type 'f'. There is no implicit conversion to type 'i'.
from /Users/victor/.rvm/gems/ruby-2.5.5@dpms-kaefer/gems/parsecs-0.8.0/lib/parsec.rb:75:in `error_check'
```

Reply:
https://www.csee.umbc.edu/courses/undergraduate/202/spring07/Lectures/ChangSynopses/modules/m04-overload/slides.php?print#:~:text=A%20function's%20signature%20includes%20the,part%20of%20a%20function's%20signature.